google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/AppEngineProjectService.java

```diff
@@ -101,4 +104,29 @@ public abstract AppEngineStandardRuntime getAppEngineStandardDeclaredRuntime(
    */
   public abstract boolean isJarOrWarMavenBuild(@NotNull Module module);
 
+  public abstract Optional<String> getServiceNameFromAppYaml(@NotNull String appYamlPath);
+
+  public abstract FlexibleRuntime getFlexibleRuntimeFromAppYaml(@NotNull String appYamlPathString);
```

Comment:
Which do you prefer: returning FlexibleRuntime.UNKNOWN or Optional.empty()?

I'm kind of tempted to avoid using Optional here...